FrontEnd/src/UserAuthentication/SignUpPage.jsx

```diff
@@ -1,77 +1,124 @@
-                        <div className="signUpPassword">
-                            <label htmlFor="password">Password</label>
-                            <input name="password" onChange={(event) => setPassword(event.target.value)} placeholder="Enter Password" />
-                        </div>
-                        <p className="linkToLogin" onClick={handleLoginRoute}>Already have an account?</p>
-                        <button className="sendSignUpButton" onClick={handleSignUp}>Sign Up</button>
-                    </form>
-                </div>
-                <div>
-                    <img className="signUpImage" src="src/Images/coworking-woman-doing-online-shopping-at-home.gif"/>
-                </div>
+          <form className="signUpForm">
+            <h3>Create an account</h3>
+            <div className="signUpFirstName">
+              <label htmlFor="firstName">First name</label>
+              <input
+                name="firstName"
+                onChange={(event) => setFirstName(event.target.value)}
+                required="required"
+                placeholder="Enter First name"
+              />
             </div>
+            <div className="signUpLastName">
+              <label htmlFor="lastName">Last name</label>
+              <input
+                name="lastName"
+                onChange={(event) => setLastName(event.target.value)}
+                placeholder="Enter Last name"
+              />
+            </div>
+            <div className="signUpEmail">
+              <label htmlFor="email">Email</label>
+              <input
+                name="email"
+                onChange={(event) => setEmail(event.target.value)}
+                placeholder="Enter Email"
+                required="required"
+              />
+            </div>
+            <div className="signUpUsername">
+              <label htmlFor="username">Username</label>
+              <input
+                name="username"
+                onChange={(event) => setUserName(event.target.value)}
+                required="required"
+                placeholder="Enter Username"
+              />
+            </div>
+            <div className="signUpPassword">
+              <label htmlFor="password">Password</label>
+              <input
+                name="password"
+                type={isPasswordVisible ? "text" : "password"}
+                onChange={(event) => setPassword(event.target.value)}
+                placeholder="Enter Password"
+              />
+              <span
+                className="icon"
+                onClick={() => setIsPasswordVisible(!isPasswordVisible)}
+              >
+                {isPasswordVisible ? (
+                  <i className="fa-solid fa-eye"></i>
+                ) : (
+                  <i className="fa-solid fa-eye-slash"></i>
+                )}
+              </span>
+            </div>
+            <p className="linkToLogin" onClick={handleLoginRoute}>
```

Comment:
Use `<Link />`? https://reactrouter.com/en/main/components/link

Reply:
I have replaced it in most places with Link. However, when it I need a page redirection inside a function, it doesn't work (for instance, redirecting to MainPage on successful login after calling the 'handleLogin' function). So I had the option of using 'useNavigate' or 'useHistory' from React so I left the useNavigate in those places.